src/main/java/com/minecolonies/coremod/commands/ActionMenu.java

```diff
@@ -63,4 +57,72 @@ public ForgePermissionNodes getForgePermissionNode()
     {
         return actionCommandClass;
     }
+
+    @Override
+    public int hashCode()
+    {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((actionArgumentList == null) ? 0 : actionArgumentList.hashCode());
+        result = prime * result + ((actionCommandClass == null) ? 0 : actionCommandClass.hashCode());
+        result = prime * result + ((description == null) ? 0 : description.hashCode());
+        result = prime * result + ((forgePermissionNode == null) ? 0 : forgePermissionNode.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 10, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)